src/libgt2/libgt2.cpp

```diff
@@ -0,0 +1,268 @@
+            pd->fetch_progress.total_deltas);
+    } else {
+        printf("net %3d%% (%4" PRIuZ " kb, %5u/%5u)  /  idx %3d%% (%5u/%5u)  /  chk %3d%% (%4" PRIuZ "/%4" PRIuZ ")%s\n",
+            network_percent, kbytes,
+            pd->fetch_progress.received_objects, pd->fetch_progress.total_objects,
+            index_percent, pd->fetch_progress.indexed_objects, pd->fetch_progress.total_objects,
+            checkout_percent,
+            pd->completed_steps, pd->total_steps,
+            pd->path);
+    }
+}
+
+static int sideband_progress(const char* str, int len, void* payload) {
+    (void)payload; /* unused */
+
+    printf("remote: %.*s", len, str);
+    fflush(stdout);
+    return 0;
+}
+
+static int fetch_progress(const git_indexer_progress* stats, void* payload) {
+    progress_data* pd = (progress_data*)payload;
+    pd->fetch_progress = *stats;
+    print_progress(pd);
+    return 0;
+}
+
+static void checkout_progress(const char* path, size_t cur, size_t tot, void* payload) {
+    progress_data* pd = (progress_data*)payload;
+    pd->completed_steps = cur;
+    pd->total_steps = tot;
+    pd->path = path;
+    print_progress(pd);
+}
+
+// Callback for clone authentication - will be used when password is not set in repo_url
+// Does not work with LFS download as it requires additional authentication when password is not set in repository url
+int cred_acquire_cb(git_credential** out,
+    const char* url,
+    const char* username_from_url,
+    unsigned int allowed_types,
+    void* payload) {
+    char *username = NULL, *password = NULL;
+    int error = -1;
+
+    fprintf(stdout, "Authentication is required for repository clone.\n");
+    if (allowed_types & GIT_CREDENTIAL_USERPASS_PLAINTEXT) {
+        const char* env_cred = std::getenv("HF_TOKEN");
+#ifdef __linux__
+        if (env_cred) {
+            username = strdup(env_cred);
+            password = strdup(username);
+        } else {
+            fprintf(stderr, "HF_TOKEN env variable is not set.\n");
+            return -1;
+        }
+#elif _WIN32
+        if (env_cred) {
+            username = _strdup(env_cred);
+            password = _strdup(username);
+        } else {
+            fprintf(stderr, "HF_TOKEN env variable is not set.\n");
+            return -1;
+        }
+#endif
+        error = git_credential_userpass_plaintext_new(out, username, password);
+        if (error < 0) {
+            fprintf(stderr, "Creating credentials failed.\n");
+            error = -1;
+        }
+    } else {
+        fprintf(stderr, "Only USERPASS_PLAINTEXT supported in OVMS.\n");
+        return 1;
+    }
+
+    free(username);
+    free(password);
+    return error;
+}
+
+// C-style callback functions section used in libgt2 library ENDS ********************************
+bool HfDownloader::isPullHfModelModeOn() {
+    return this->pullHfModelMode;
+}
+
+bool HfDownloader::CheckIfProxySet() {
+    const char* envCred = std::getenv("https_proxy");
```

Comment:
We still do have get env in HfDownloader

Reply:
Removed today.